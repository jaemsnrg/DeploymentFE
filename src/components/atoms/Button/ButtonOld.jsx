import React from 'react';
import PropTypes from 'prop-types';

import c from './button.scss';

// props
// buttonText [*] - determines text value of button
// buttonWidth [*] - set width of button in px, % etc
// buttonColor [blue, white] - determines fill theme of button
// func [*] - sets function to call when button is clicked
// tall - adds extra height padding around the text.
// noCaps - Provide as true if you don't want the button text to be all capitalised.
// link - A boolean indicating whether or not the button is a link.
// href - Used as the link location for a link represented by a button.
// withShadow - A boolean indicating whether or not the button should cast a shadow.
// small - A stylistic prop that allows us to produce buttons with smaller text that is not bold.

const Button = (props) => {
  const styles = {
    width: props.buttonWidth,
  };

  const classes = [props.small ? 'text__bs--medium' : 'text__bm--bold', c.standardButton];

  if (props.small) {
    classes.push(c['standardButton--small'], c['materialIconSize--small']);
  }
  props.tall === true ? classes.push(c['standardButton--tall']) : null;
  props.short === true ? classes.push(c['standardButton--short'], 'text__searchInput') : null;

  switch (props.buttonColor) {
    case 'white':
      classes.push(c.white);
      break;
    case 'whiteNoBorder':
      classes.push(c.whiteNoBorder);
      break;
    case 'blue':
      classes.push(c.blue);
      break;
    case 'whiteBlue':
      classes.push(c.whiteBlue);
    case 'blueBlue':
      classes.push(c.blueBlue);
      break;
  }

  if (props.disabled && props.disabledStyle === 'form') {
    classes.push(c['form--disabled']);
  }

  if (props.disabled && props.disabledStyle === 'standard') {
    classes.push(c['standard--disabled']);
  }

  if (props.withShadow) {
    classes.push(c['standardButton--withShadow']);
  }

  if (props.link) {
    return (
      <a
        onClick={props.func}
        style={styles}
        className={classes.join(' ')}
        id={props.mixpanelId}
        href={props.href}
      >
      <span className="material-icons">{props.materialIconCode}</span> 
        {props.noCaps ? props.buttonText : props.buttonText.toUpperCase()}
      </a>
    );
  }
  return (
    <button disabled={props.disabled} onClick={props.func} style={styles} className={classes.join(' ')} id={props.mixpanelId}>
      <span className="material-icons">{props.materialIconCode}</span> 
      {props.noCaps ? props.buttonText : props.buttonText.toUpperCase()}
    </button>
  );
};

Button.defaultProps = {
  tall: true,
};

Button.propTypes = {
  tall: PropTypes.bool,
  short: PropTypes.bool,
  buttonColor: PropTypes.string,
  buttonText: PropTypes.string,
  noCaps: PropTypes.bool,
  func: PropTypes.func,
  link: PropTypes.bool,
  mixpanelId: PropTypes.string,
  withShadow: PropTypes.bool,
  small: PropTypes.bool,
  buttonWidth: PropTypes.string,
  disabled: PropTypes.bool,
  disabledStyle: PropTypes.string,
  materialIconCode: PropTypes.string,
};

export default Button;
