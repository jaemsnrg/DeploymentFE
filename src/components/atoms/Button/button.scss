// @import "~styles/_mixins";
// @import "~styles/_colours";
// @import "~styles/_variables";

.button:hover {
}
.standardButton {
  width: 100%;
  display: inline-block;
  border-radius: 25px;
  padding: 6px;
  text-align: center;
  cursor: pointer;
  user-select: none;
  outline: none;
  @include atTablet {
    padding: 10px;
  }
  &--tall {
    padding: 10px;
    @include atTablet {
      padding: 14px;
    }
  }
  &--short {
    padding: 9px 22px;
    @include atTablet {
      padding: 9px 22px;
    }
  }
  &--small {
    padding: 7px 18px;
    @include atTablet {
      padding: 7px 18px;
    }
  }
  &--withShadow {
    border: none;
    box-shadow: 0 2px 6px 0 rgba(0, 0, 0, 0.5);
  }
}
.blue {
  background-color: $blue-dark;
  color: $white;
  border: 0.5px solid $blue-dark;
  &:hover {
    background-color: $blue;
    border: 0.5px solid $blue;
  }
}
.white {
  background-color: $white;
  color: $black;
  border: 0.5px solid $black;
  &:hover {
    color: $blue-dark;
    border: 0.5px solid $blue-dark;
  }
}
.whiteNoBorder {
  background-color: $white;
  color: $black;
  border: none;
  &:hover {
    color: $blue-dark;
    border: none;
  }
}
.whiteBlue {
  background-color: $white;
  color: $blue-dark;
  border: 0.5px solid $white;
}

.blueBlue {
  background-color: $white;
  color: $blue-dark;
  border: 0.5px solid $blue-dark;
}

.standard--disabled{
  background-color: $white;
  border-color: $black;
  color: $black;
  &:hover{
    background-color: $white;
    border-color: $black;
    color: $black;
  }
}

.form--disabled{
  cursor: default;
  background-color: $grey-light;
  color: $grey-dark;
  border-color: $grey-light;
  &:hover{
    background-color: $grey-light;
    color: $grey-dark;
    border-color: $grey-light;
  }
}

.materialIconSize{
  &--small {
    span{
      position: relative;
      top: 3px;
      font-size: 13px;
      margin-right: 2px;
      @media (min-width: #{$tablet-width}){
        font-size: 15px;
      }
    }
  }
}