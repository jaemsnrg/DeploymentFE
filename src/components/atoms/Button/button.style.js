import styled from 'styled-components';

export const ButtonOuter = styled.button`
  width: 100%;
  display: inline-block;
  border-radius: 25px;
  text-align: center;
  cursor: pointer;
  user-select: none;
  outline: none;
  padding: ${({ padding }) => (padding ? padding : '4px')};
  ${({ theme, fontSize }) => (fontSize ? theme.text[fontSize] : theme.text.bs)};
  color: ${({ buttonColor, theme }) =>
    buttonColor === 'inverted' ? theme.colours.blueDark : 'white'};
  border: 0.5px solid ${({ theme }) => theme.colours.blueDark};
  background-color: ${({ buttonColor, theme }) =>
    buttonColor === 'inverted' ? 'white' : theme.colours.blueDark};
  opacity: ${({ opaque }) => (opaque ? 0.5 : 1)};
  font-weight: ${({ buttonFontWeight }) => (buttonFontWeight === 'bold' ? 'bold' : 'normal')};
  letter-spacing: ${({ letterSpacing }) => letterSpacing};
`;
/*

${({tall, theme}) => {
    try {
        if (tall) {
            return `padding: 10px;`
        }
    } catch (error) {}
}};

${({theme}) => theme.mq.atTablet`
    ${({tall, theme}) => {
        try {
            if (tall) {
                return `padding: 14px;`
            }
        } catch (error) {}
    }};

`}

${(props) => props.theme.mq.atDesktop`
    ${({tall, theme}) => {
        try {
            if (tall) {
                return theme.text.bxlMedium
            }
        } catch (error) {}
    }};

`}

*/
