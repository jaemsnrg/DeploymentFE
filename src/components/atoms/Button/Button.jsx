import React from 'react';
import PropTypes from 'prop-types';

import { ButtonOuter } from './button.style';
// import { AnimatedElipsis } from '@atoms';

// props
// buttonText [*] - determines text value of button
// buttonWidth [*] - set width of button in px, % etc
// buttonColor [blue, white] - determines fill theme of button
// func [*] - sets function to call when button is clicked
// tall - adds extra height padding around the text.
// noCaps - Provide as true if you don't want the button text to be all capitalised.
// link - A boolean indicating whether or not the button is a link.
// href - Used as the link location for a link represented by a button.
// withShadow - A boolean indicating whether or not the button should cast a shadow.
// small - A stylistic prop that allows us to produce buttons with smaller text that is not bold.

const Button = ({
  buttonText,
  buttonColor,
  buttonFontWeight,
  func,
  link,
  buttonWidth,
  loading,
  opaque,
  slim,
  letterSpacing,
  fontSize,
  padding,
}) => {
  let buttonContent = null;
  if (loading) {
    // buttonContent = <AnimatedElipsis />;
  } else if (buttonText) {
    buttonContent = buttonText;
  }

  return (
    <ButtonOuter
      onClick={func ? func : null}
      fontSize={fontSize}
      buttonFontWeight={buttonFontWeight}
      buttonColor={buttonColor}
      opaque={opaque}
      letterSpacing={letterSpacing}
      buttonFontWeight={buttonFontWeight}
      padding={padding}
    >
      {buttonContent}
      {/* <span className="material-icons">{props.materialIconCode}</span>  */}
      {/* {props.noCaps ? props.ButtonText : props.buttonText.toUpperCase()} */}
    </ButtonOuter>
  );
};

Button.propTypes = {
  // tall: PropTypes.bool,
  // short: PropTypes.bool,
  padding: PropTypes.string,
  slim: PropTypes.bool,
  buttonColor: PropTypes.string,
  buttonText: PropTypes.string,
  buttonFontWeight: PropTypes.string,
  letterSpacing: PropTypes.string,
  fontSize: PropTypes.string,
  // noCaps: PropTypes.bool,
  func: PropTypes.func,
  link: PropTypes.bool,
  // mixpanelId: PropTypes.string,
  // withShadow: PropTypes.bool,
  // small: PropTypes.bool,
  buttonWidth: PropTypes.string,
  loading: PropTypes.bool,
  opaque: PropTypes.bool,
  // disabled: PropTypes.bool,
  // disabledStyle: PropTypes.string,
  // materialIconCode: PropTypes.string,
};

export default Button;
