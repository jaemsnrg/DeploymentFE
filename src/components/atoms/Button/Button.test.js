import React from 'react';
import { shallow } from 'enzyme';
import mountWithTheme from '@testUtils/mountWithTheme';
import mockTheme from '@testUtils/mockTheme';

import Button from './Button';

describe('Button', () => {
  it('should render without any errors', () => {
    const TestContainer = () => <Button buttonText="Answer" />;
    const wrapper = shallow(<TestContainer />);
    expect(wrapper.find(Button).length).toBe(1);
  });

  it('should contain the correct text', () => {
    const wrapper = mountWithTheme(<Button buttonText="Answer" />, mockTheme);
    expect(wrapper.contains('Answer')).toBe(true);
  });

  it('should animate when specified', () => {
    const wrapper = mountWithTheme(<Button loading={true} />, mockTheme);
    expect(wrapper.contains('\u25CF ')).toBe(true);
  });

});