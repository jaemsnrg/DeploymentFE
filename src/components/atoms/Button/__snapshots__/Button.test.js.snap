// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches the snapshot 1`] = `
<button
  className="  "
  style={
    Object {
      "width": undefined,
    }
  }
>
  TEST
</button>
`;
